magefile.go

```diff
@@ -1295,3 +1320,144 @@ func shouldBuildAgent() bool {
 	}
 	return ret
 }
+
+// Pre-requisite: user must have the gcloud CLI installed
+func authGCP(ctx context.Context) error {
+	// Check if gcloud CLI is installed
+	const cliName = "gcloud"
+	cmd := exec.CommandContext(ctx, "which", cliName)
+	if err := cmd.Run(); err != nil {
+		return fmt.Errorf("%s CLI is not installed: %w", cliName, err)
+	}
+
+	// Check if user is already authenticated
+	var authList []struct {
+		Account string `json:"account"`
+	}
+	for authSuccess := false; !authSuccess; {
+		cmd = exec.CommandContext(ctx, cliName, "auth", "list", "--filter=status:ACTIVE", "--format=json")
+		output, err := cmd.Output()
+		if err != nil {
+			return fmt.Errorf("unable to list authenticated accounts: %w", err)
+		}
+
+		if err := json.Unmarshal(output, &authList); err != nil {
+			return fmt.Errorf("unable to parse authenticated accounts: %w", err)
+		}
+
+		if len(authList) > 0 {
+			// We have at least one authenticated, active account. All set!
+			authSuccess = true
+			continue
+		}
+
+		fmt.Fprintln(os.Stderr, "❌  GCP authentication unsuccessful. Retrying...")
+
+		// Try to authenticate user
+		cmd = exec.CommandContext(ctx, cliName, "auth", "login")
+		if err := cmd.Run(); err != nil {
+			return fmt.Errorf("unable to authenticate user: %w", cliName, err)
+		}
+	}
+
+	// Check that authenticated account's email domain name
+	const expectedEmailDomain = "elastic.co"
+	email := authList[0].Account
+	parts := strings.Split(email, "@")
+	if len(parts) != 2 || parts[1] != expectedEmailDomain {
+		return fmt.Errorf("please authenticate with your @%s email address (currently authenticated with %s)", expectedEmailDomain, email)
+	}
+
+	// Check the authenticated account's project
+	cmd = exec.CommandContext(ctx, cliName, "config", "get", "core/project")
+	output, err := cmd.Output()
+	if err != nil {
+		return fmt.Errorf("unable to get project: %w", err)
+	}
+
+	const expectedProject = "elastic-obs-integrations-dev"
```

Comment:
I believe this should be `elastic-platform-ingest`.

Reply:
I agree, we should use elastic-platform-ingest

Reply:
Thanks, will change.  @pierrehilbert How do I get access to this project, so I can test this PR?

Reply:
@ycombinator you need to be part of the ingest-team google group which is already the case.